File: tests/go/mathops/random_math_operations_test.go
package mathops

import (
	"fmt"
	"testing"
)

func TestGenerateRandomOddNumber(t *testing.T) {
	r := &RandomMathOperations{}
	for i := 1; i <= 20; i++ {
		t.Run(fmt.Sprintf("Run_%d/20", i), func(t *testing.T) {
			number := r.GenerateRandomOddNumber()
			if number < 1 || number > 100 {
				t.Errorf("number %d not in range [1, 100]", number)
			}
			if number%2 != 1 {
				t.Errorf("number %d should be odd", number)
			}
		})
	}
}

func TestGenerateRandomEvenNumber(t *testing.T) {
	r := &RandomMathOperations{}
	for i := 1; i <= 20; i++ {
		t.Run(fmt.Sprintf("Run_%d/20", i), func(t *testing.T) {
			number := r.GenerateRandomEvenNumber()
			if number < 0 || number > 100 {
				t.Errorf("number %d not in range [0, 100]", number)
			}
			if number%2 != 0 {
				t.Errorf("number %d should be even", number)
			}
		})
	}
}

func TestGenerateRandomPrimeCandidate(t *testing.T) {
	r := &RandomMathOperations{}
	for i := 1; i <= 20; i++ {
		t.Run(fmt.Sprintf("Run_%d/20", i), func(t *testing.T) {
			number := r.GenerateRandomPrimeCandidate()
			if number < 2 || number > 200 {
				t.Errorf("number %d not in range [2, 200]", number)
			}
			if !IsPrime(number) {
				t.Errorf("number %d should be prime", number)
			}
		})
	}
}

func TestIsPrime(t *testing.T) {
	t.Run("prime numbers", func(t *testing.T) {
		primes := []int{2, 3, 5, 7, 11, 97}
		for _, p := range primes {
			if !IsPrime(p) {
				t.Errorf("IsPrime(%d) = false, want true", p)
			}
		}
	})

	t.Run("non-prime numbers", func(t *testing.T) {
		nonPrimes := []int{1, 4, 6, 9, 100}
		for _, n := range nonPrimes {
			if IsPrime(n) {
				t.Errorf("IsPrime(%d) = true, want false", n)
			}
		}
	})
}

File: typescript/tests/jest/randomMathOperations.test.ts
import { RandomMathOperations } from '../../src/randomMathOperations';

/**
 * Test suite for RandomMathOperations.
 *
 * - generateRandomOddNumber() test always passes
 * - generateRandomEvenNumber() test always passes
 * - generateRandomPrimeCandidate() test always passes
 */
describe('RandomMathOperations Tests', () => {
  const randomOps = new RandomMathOperations();

  const runs = Array.from({ length: 20 }, (_, i) => i + 1);

  describe.each(runs)('Run %i/20', (_run) => {
    test('Random odd number should be odd', () => {
      const number = randomOps.generateRandomOddNumber();
      expect(number).toBeGreaterThanOrEqual(1);
      expect(number).toBeLessThanOrEqual(100);
      expect(number % 2).toBe(1);
    });

    test('Random even number should be even', () => {
      const number = randomOps.generateRandomEvenNumber();
      expect(number).toBeGreaterThanOrEqual(0);
      expect(number).toBeLessThanOrEqual(100);
      expect(number % 2).toBe(0);
    });

    test('Random prime candidate should be prime', () => {
      const number = randomOps.generateRandomPrimeCandidate();
      expect(number).toBeGreaterThanOrEqual(2);
      expect(number).toBeLessThanOrEqual(200);
      expect(RandomMathOperations.isPrime(number)).toBe(true);
    });
  });

  test('Helper method isPrime should correctly identify prime numbers', () => {
    // Test prime numbers
    expect(RandomMathOperations.isPrime(2)).toBe(true);
    expect(RandomMathOperations.isPrime(3)).toBe(true);
    expect(RandomMathOperations.isPrime(5)).toBe(true);
    expect(RandomMathOperations.isPrime(7)).toBe(true);
    expect(RandomMathOperations.isPrime(11)).toBe(true);
    expect(RandomMathOperations.isPrime(97)).toBe(true);

    // Test non-prime numbers
    expect(RandomMathOperations.isPrime(1)).toBe(false);
    expect(RandomMathOperations.isPrime(4)).toBe(false);
    expect(RandomMathOperations.isPrime(6)).toBe(false);
    expect(RandomMathOperations.isPrime(9)).toBe(false);
    expect(RandomMathOperations.isPrime(100)).toBe(false);
  });
});
